src/nvim/map.c

```diff
@@ -123,29 +123,8 @@
     kh_clear(T##_##U##_map, map->table); \
   }
 
-static inline khint_t String_hash(String s)
```

Comment:
@bfredl From `map.c`? There is a point to delete `map.[ch]` entirely, ineffective third hash interface which just indirects second hash interface which uses second hash implementation is not a thing we really need. I was expressing the reasons previously.

Initially I did not delete String_hash, I moved it to `dispatch.c`. It was then deleted when I decided that dispatch is a perfect case for gperf. If you think it can be needed I can move the function to `khash.h`, but I do not see any reasons to keep anything in `map.c` I can remove instead.